backends/dpdk/dpdkArch.cpp

```diff
@@ -2249,6 +2249,13 @@ void CollectAddOnMissTable::postorder(const IR::P4Table* t) {
     if (use_add_on_miss) {
         for (auto action : t->getActionList()->actionList) {
             auto action_decl = refMap->getDeclaration(action->getPath())->to<IR::P4Action>();
+            // Map the compiler generated internal name of action (emitted in .spec file) with
+            // user visible name in P4 program.
+            // To get the user visible name, strip any prefixes from externalName.
+            cstring userVisibleName = action_decl->externalName();
+            userVisibleName = userVisibleName.findlast('.');
+            userVisibleName = userVisibleName.trim(".\t\n\r");
```

Comment:
do we really expect spaces in this name?

Reply:
Not really! I can keep just '.' in this list of characters to be trimmed.
I will take care of this along with other changes in a future PR.

Reply:
Not really! I can keep just '.' in this list of characters to be trimmed.
I will take care of this along with other changes in a future PR.